### DISCUSSION TITLE
convert: improper image header

### DISCUSSION THREAD

**Discussion by dhaval1671** [ImageMagick/ImageMagick#4193] [Help]
### Description
Using convert command line tool to convert tiff to PNG and facing below given error
convert: improper image header `/Alftemp/MRK_16.tif' @ **error/tiff.c/ReadTIFFImage/1440.**

### Steps to Reproduce

**[tomcat@tmng-alfresco6-1 ~]$ convert  -quiet    /Alftemp/MRK_16.tif /Alftemp/16.png
convert: improper image header `/Alftemp/MRK_16.tif' @ error/tiff.c/ReadTIFFImage/1440.
convert: no images defined `/Alftemp/16.png' @ error/convert.c/ConvertImageCommand/3285.**

### System Configuration

- ImageMagick version: Version: ImageMagick 7.0.10-26 Q16 x86_64 2020-08-10 https://imagemagick.org
- Environment (Operating system, version and so on): RHEL 7
- Additional information: TIFF -> PNG 

[MRK_16 (1).zip](https://github.com/ImageMagick/ImageMagick/files/5378784/MRK_16.1.zip)

**Comment by t246246** [comment]
Your tiff file does not have some required field Photometric Interpretation, which has no default value.  https://www.awaresystems.be/imaging/tiff/tifftags/photometricinterpretation.html When I tried to open your file in Gimp, it says:

Could not get photometric from 'MRK_16.tif'. Image is CCITT compressed, assuming min-is-white

However, imagemagick do not assume it, so got an error.

**Comment by dhaval1671** [comment]
Thanks for your response but it also failing with this

[tomcat@tmng-alfresco6-1 ~]$ **convert  -define  quantum:polarity=photometric-interpretation=min-is-white    /Alftemp/MRK_16.tif /Alftemp/16.png**
convert: improper image header `/Alftemp/MRK_16.tif' @ error/tiff.c/ReadTIFFImage/1440.
convert: no images defined `/Alftemp/16.png' @ error/convert.c/ConvertImageCommand/3285.

**Comment by t246246** [comment]
I don't know whether that -define should work or not.
If you have tiffset command comes from libtiff,

% tiffset -s 262 0 MRK_16.tif 

set the value, then you can convert it with imagemagick.

**Comment by dhaval1671** [comment]
Any fix with convert?

**Comment by dhaval1671** [comment]
I also tried to use set with convert but result is same

[tomcat@tmng-alfresco6-1 ~]$ convert -set 262 0 /Alftemp/MRK_16.tif /Alftemp/16.png
convert: improper image header `/Alftemp/MRK_16.tif' @ error/tiff.c/ReadTIFFImage/1440.

**Comment by t246246** [comment]
I don't think this would be fixed.  In short, it is not necessary to handle broken file.

-define quantum:polarity=photometric-interpretation=min-is-white should be -define quantum:polarity=min-is-white, anyway that does not work.  It seems that -define is applied to output file.

IMHO, it is not a bad idea to provide some option that can supply missing information, or overwrite the value of input file when reading though.

**Comment by dhaval1671** [comment]
You know what same image is working fine in old version of Magic - Version: ImageMagick 6.7.8-9 2018-12-11 Q16 http://www.imagemagick.org

its workingwith the given version

**Comment by t246246** [comment]
As far as I understand,  CCITT Group4 compression does not require min-is-white.
Your image can be interpreted 2 ways.  So I think the current behavior is right.
If you do not think so,  you need to show the specification of tiff or compression to persuade the developer.

**Reply by kjsharke** [reply]
https://web.archive.org/web/20210108174645/https://www.adobe.io/content/dam/udp/en/open/standards/tiff/TIFF6.pdf p50

> _PhotometricInterpretation_. An encoded CCITT string is self-photometric, defined in terms of white and black runs. Yet TIFF defines a tag called PhotometricInterpretation that also purports to define what is white and what is black. Somewhat arbitrarily, we adopt the following convention:
> 
> The “normal” PhotometricInterpretation for bilevel CCITT compressed data is WhiteIsZero. In this case, the CCITT “white” runs are to be interpretated as white, and the CCITT “black” runs are to be interpreted as black. However, if the PhotometricInterpretation is BlackIsZero, the TIFF reader must reverse the meaning of white and black when displaying and printing the image.
